src/Forwarder.cpp

```diff
@@ -307,14 +307,21 @@ void Forwarder::pushConverterToStream(ConverterSettings const &ConverterInfo,
   }
 
   URI TopicURI;
-  if (!Uri.host.empty()) {
-    TopicURI.host = Uri.host;
+  if (!Uri.HostPort.empty()) {
```

Comment:
All this URI stuff should be in a separate method, e.g. createTopicUrl.
From the name it isn't clear to me that Uri is the broker uri, so perhaps it should be renamed to BrokerUri.